src/containers/RedashContainer/index.jsx

```diff
@@ -59,23 +60,30 @@ class RedashContainer extends Component {
   }
 
   async fetchSetState({ dataKeyIdentifier, redashDataUrl }) {
-    const redashData = await fetchJson(redashDataUrl);
+    try {
+      this.setState({ isLoading: true });
+      const redashData = await fetchJson(redashDataUrl);
 
-    this.setState({
-      datasets: telemetryDataToDatasets(redashData, dataKeyIdentifier),
-    });
+      this.setState({
+        datasets: telemetryDataToDatasets(redashData, dataKeyIdentifier),
+      });
+      this.setState({ isLoading: false });
```

Comment:
Same comment as above - combine state updates as much as possible.

Reply:
Same as before, but the `isLoading: false` into the finally clause